docs/dg/dev/upgrade-and-migrate/integrate-api-platform.md

```diff
@@ -0,0 +1,333 @@
+```
+
+{% info_block infoBox "Module placeholder" %}
+
+The exact module versions will be provided in the final documentation. The above serves as a placeholder for the module list.
+
+{% endinfo_block %}
+
+## 2. Register bundles
+
+Register the required bundles in your application's bundle configuration files for each application layer where you want to enable API Platform.
+
+### For Glue application
+
+`config/Glue/bundles.php`
+
+```php
+<?php
+
+declare(strict_types = 1);
+
+use ApiPlatform\Symfony\Bundle\ApiPlatformBundle;
+use Nelmio\CorsBundle\NelmioCorsBundle;
+use Spryker\ApiPlatform\SprykerApiPlatformBundle;
+use Symfony\Bundle\FrameworkBundle\FrameworkBundle;
+use Symfony\Bundle\TwigBundle\TwigBundle;
+
+return [
+    FrameworkBundle::class => ['all' => true],
+    TwigBundle::class => ['all' => true],
+    NelmioCorsBundle::class => ['all' => true],
+    ApiPlatformBundle::class => ['all' => true],
+    SprykerApiPlatformBundle::class => ['all' => true],
+];
+```
+
+### For GlueStorefront application
+
+`config/GlueStorefront/bundles.php`
+
+```php
+<?php
+
+declare(strict_types = 1);
+
+use ApiPlatform\Symfony\Bundle\ApiPlatformBundle;
+use Nelmio\CorsBundle\NelmioCorsBundle;
+use Spryker\ApiPlatform\SprykerApiPlatformBundle;
+use Symfony\Bundle\FrameworkBundle\FrameworkBundle;
+use Symfony\Bundle\TwigBundle\TwigBundle;
+
+return [
+    FrameworkBundle::class => ['all' => true],
+    ApiPlatformBundle::class => ['all' => true],
+    SprykerApiPlatformBundle::class => ['all' => true],
+    TwigBundle::class => ['all' => true],
+    NelmioCorsBundle::class => ['all' => true],
+];
+```
+
+### For GlueBackend application
+
+`config/GlueBackend/bundles.php`
+
+```php
+<?php
+
+declare(strict_types = 1);
+
+use ApiPlatform\Symfony\Bundle\ApiPlatformBundle;
+use Nelmio\CorsBundle\NelmioCorsBundle;
+use Spryker\ApiPlatform\SprykerApiPlatformBundle;
+use Symfony\Bundle\FrameworkBundle\FrameworkBundle;
+use Symfony\Bundle\TwigBundle\TwigBundle;
+
+return [
+    FrameworkBundle::class => ['all' => true],
+    ApiPlatformBundle::class => ['all' => true],
+    SprykerApiPlatformBundle::class => ['all' => true],
+    TwigBundle::class => ['all' => true],
+    NelmioCorsBundle::class => ['all' => true],
+];
+```
+
+## 3. Configure API Platform
+
+Create configuration files for the API Platform in each application layer.
+
+### Configure for Glue
+
+`config/Glue/packages/spryker_api_platform.php`
+
+```php
+<?php
+
+declare(strict_types=1);
+
+use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;
+
+return static function (ContainerConfigurator $containerConfigurator): void {
+    $containerConfigurator->extension('spryker_api_platform', [
+        // Configure which API types to generate and cache warm
+        // Common values: 'backend', 'storefront', 'merchant-portal'
+        'api_types' => [
+            'storefront',
+        ],
+
+        // Optional: Configure source directories to search for schema files
```

Comment:
could you please explain why this might be needed? Reference to Symfony documentation is also fine